### PR TITLE
[fix]: [PIE-5788]: Improved generic error message

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Improved Error Message by adding appropriate message when some fields are not provided for runtime inputs.

Issue: when we declare env/srvc as runtime input and don't provide its value in inputSet, during its preflightCheck we didn't appropriate message.
Change: have improved the error message and also handled a NullNode Case also which may occur around this(although not reported in the ticket).
Testing: Expected behaviour on local. Previous preflight flows are intact, confirmed by running automation against my branch
Screenshot 2022-12-02 at 12 14 41 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40919)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger messagecheck

File: 878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java [hunk 1]
```diff
@@ -56,6 +57,9 @@ public List<EntityDetail> extractInputReferredEntityFromYaml(String accountIdent
         entityDetails.add(referredUsage.getReferredEntity());
       } else if (fqnToObjectMapMergedYaml.containsKey(fqn)) {
         Object finalNode = fqnToObjectMapMergedYaml.get(fqn);
+        if (finalNode instanceof NullNode) {
+          throw new InvalidRequestException("Some fields are not provided!");
```
Comment (naidu-harness) [inline 878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java:61]: is there any way to send what exact fields are not provided? This message will be very baugue

File: 970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java [hunk 2]
```diff
@@ -194,7 +195,11 @@ public static boolean containsInputSetValidator(String value) {
       ParameterField<?> parameterField = YamlPipelineUtils.read(value, ParameterField.class);
       return parameterField.getInputSetValidator() != null;
     } catch (IOException e) {
-      throw new InvalidRequestException(value + " is not a valid value for runtime input");
+      if (EmptyPredicate.isEmpty(value)) {
+        throw new InvalidRequestException("Some runtime input fields are not provided!");
```
Comment (naidu-harness) [inline 970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java:199]: ditto

File: 878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java (hunk 1 above)
Comment (naidu-harness) [reply 878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java:61]: may be provide fqn itself if final node is null?

**Review by archit-harness — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40919)* status: 2 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/40919#-NIChZJjhgIpX0MYZVLt-r2-61:-NIGCo2zBvrcwV4vd9cQ:bv9ntuw) ([raw file](https://github.com/harness/harness-core/blob/faf6b0157250e106f2383a9ba0332953a12d0136/878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java#L61)):*
<details><summary><i>Previously, naidu-harness (Sanyasi Naidu Annepu) wrote…</i></summary><blockquote>

may be provide fqn itself if final node is null?
</blockquote></details>

@senjucanon2 we have fqn string available here, thus we can use in the exception
___
*[`970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java` line 199 at r2](https://reviewable.io/reviews/harness/harness-core/40919#-NIBZGIiZDTPh4EeKhGP-r2-199:-NIGCtZw3YnZMoYH7I1c:b-8qsmz0) ([raw file](https://github.com/harness/harness-core/blob/faf6b0157250e106f2383a9ba0332953a12d0136/970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java#L199)):*
<details><summary><i>Previously, naidu-harness (Sanyasi Naidu Annepu) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Same we can pass fqn, i think the change is small only

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40919)* status: 2 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java` line 61 at r2](https://reviewable.io/reviews/harness/harness-core/40919#-NIChZJjhgIpX0MYZVLt-r2-61:-NIGa-YwCbluR5FjCjUv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/faf6b0157250e106f2383a9ba0332953a12d0136/878-ng-common-utilities/src/main/java/io/harness/PipelineSetupUsageUtils.java#L61)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@senjucanon2 we have fqn string available here, thus we can use in the exception
</blockquote></details>

Done.
___
*[`970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java` line 199 at r2](https://reviewable.io/reviews/harness/harness-core/40919#-NIBZGIiZDTPh4EeKhGP-r2-199:-NIGa0Ts93u4vvo3_XHV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/faf6b0157250e106f2383a9ba0332953a12d0136/970-ng-commons/src/main/java/io/harness/pms/yaml/ParameterField.java#L199)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Same we can pass fqn, i think the change is small only
</blockquote></details>

Done.

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40919)* status: 2 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40919)* status: 2. unresolved discussions.

**Review by naidu-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40919)* status: :shipit: complete! ✓

**Pull request merged**